### DISCUSSION TITLE
Installing in a partition

### DISCUSSION THREAD

**Discussion by dbarronoss** [helloSystem/hello#347] [General]
I'm one of those distro collector guys, having like 7 favorite distros installed and I'm branching out to *BSD.  Having used GhostBSD satisfactoily for a few days, I actually think I prefer the Hello system (only used on LiveBoot), and would like to install it to my HD, however, I do not wish to sacrifice the entire disk to do so.
Point me in a direction, please.

**Comment by probonopd** [comment]
Hello @dbarronoss. I have not implemented this kind of functionality yet, because systems with multiple operating systems on the same disk can end up in a non-bootable state rather fast and I don't have the resources to test all possible combinations of operating systems. Sorry.

**Reply by dbarronoss** [reply]
In my case, I just boot from ReFind, so the OS is not required to do any EFI management or grub or anything like that.  However, for the greater masses....
I suppose I could just dd the LIVE environment onto a partition and then grow the partition to the full size.  That's a lot of trouble if something easier might be coming fairly soon (see next thread).

**Comment by timmrogers** [comment]
Something like VMware or VirtualBox seems like the obvious solution to this
use case. I have 21 OSs (including helloSystem) available on my iMac. Works
great for checking out and keeping up with various operating systems.

On Fri, Mar 11, 2022 at 10:36 AM probonopd ***@***.***> wrote:

> Hello @dbarronoss <https://github.com/dbarronoss>. I have not implemented
> this kind of functionality yet, because systems with multiple operating
> systems on the same disk can end up in a non-bootable state rather fast and
> I don't have the resources to test all possible combinations of operating
> systems. Sorry.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/helloSystem/hello/discussions/347#discussioncomment-2341262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACAPGHG273FRAHDIY5S2SG3U7OHEBANCNFSM5QOEAI7Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Tim Rogers*
+1 541-221-5834
Secure email: ***@***.***

**Reply by probonopd** [reply]
...if you have lots of RAM and are fine with sacrificing some performance, then yes.

**Reply by probonopd** [reply]
But then, helloSystem has a Live ISO to try it out and is otherwise meant as the main operating system (e.g., for Macs that don't get macOS updates anymore).

**Reply by dbarronoss** [reply]
Yeah, I favor a real install (vs a virtual one) to actually assess a distro/OS/etc.  I hope that you will implement install on a partition soon, I read that rumours of 1.0 release is coming soon?

**Reply by probonopd** [reply]
If some developers will join the effort, then it is going to be much faster for sure :-)

**Reply by dbarronoss** [reply]
Are you a one-man show at moment?  I applaud your efforts, don't burn yourself out.
It's been nearly ten years since i wrote any code, was never that good at it anyway (hacker describes me better than programmer) or administrator.  I'd help in any way I could (my best efforts would be QA I would think), unless you have PHP or Perl or (wow long time since I touched that) C needs.

**Reply by probonopd** [reply]
Do you speak Python?

**Reply by dbarronoss** [reply]
Err, can't say that I speak that snake tongue, but picking up a new language is relatively easy when you've learned multiple over your lifetime.  As I say, I was never a great programmer, but I can try...I have no involvement in any project at the present time and what better way to learn BSD than to work with it.  I did manage to get a Master's in Computer Science back in the last century (199x), but I don't think it was on the strength of my programming (*lol*).

**Comment by dbarronoss** [comment]
if you want me to try to help, well I will :)  (figured should take this off the board)Looking at Python, it looks quite similar to several of the other languages I know (Pascal for one), I don't think it would be hard to be minimally competent in it.What sort of tasks would be helpful to you to assign for me (remember I'm cutting baby python teeth)?Btw, my name is Danny

    On Friday, March 11, 2022, 01:19:18 PM CST, probonopd ***@***.***> wrote:  
 
 


Do you speak Python?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by probonopd** [reply]
Hi Danny, thanks for offering your help. Check out the "Under Construction" folder in Applications. maybe you'd like to adopt one of those?

Disk Utility would need most work, as it's currently pretty non-functional. https://github.com/helloSystem/Utilities/tree/master/System/Format%20Disk.app has some of the code needed. Does this help to get started?

Feel free to ask questions, I've been learning FreeBSD and PyQt along the way, too :-)

**Comment by dbarronoss** [comment]
No promises, but I'd love to give it a try.  Unfortunately I have a once a year visit from my best friend coming up over the next week, so will be mostly occupied (versus the rest of my life as I'm single) *lol*.Thanks for the direction.
    On Saturday, March 12, 2022, 02:52:04 PM CST, probonopd ***@***.***> wrote:  
 
 


Hi Danny, thanks for offering your help. Check out the "Under Construction" folder in Applications. maybe you'd like to adopt one of those? Disk Utility would need most work, as it's currently pretty non-functional.

Checking out https://github.com/helloSystem/Utilities/tree/master/System/Format%20Disk.app has some of the code needed. Does this help to get started? Feel free to ask questions, I've been learning FreeBSD and PyQt along the way, too :-)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
